src/core/Track.cpp

```diff
@@ -622,14 +624,10 @@ void Track::toggleSolo()
 				track->m_soloModel.setValue(false);
 			}
 		}
-		else if (!soloBefore)
+		// Revert all tracks to pre-solo values
+		else if (!otherTrackAlreadySolo)
```

Comment:
Thinking out loud:

`soloEnabled` is true if *this* track is solo . `otherTrackAlreadySolo` is true if *any* track is solo. At first glance it seems like this function is only called when the solo LED is clicked. So you may think that if you uncheck the solo LED of a track, then `otherTrackAlreadySolo` must be false. *But* this function also calls itself through this line:

    track->m_soloModel.setValue(false);

And in that case a solo LED is unchecked while another solo LED is active. A bit confusing and ineffective but probably the simplest solution.

TL;DR there is nothing wrong here.